tests/manage/test_pvc_disruptive.py

```diff
@@ -0,0 +1,119 @@
+import pytest
+import logging
+from ocs_ci.ocs import constants
+from tests import helpers, disruption_helpers
+from ocs_ci.framework.testlib import ManageTest, tier4
+from tests.fixtures import (
+    create_rbd_storageclass, create_ceph_block_pool, create_cephfs_storageclass,
+    create_rbd_secret, create_cephfs_secret
+)
+
+logger = logging.getLogger(__name__)
+
+DISRUPTION_OPS = disruption_helpers.Disruptions()
+
+
+class BaseDisruption(ManageTest):
+    """
+    Base class for PVC related disruption tests
+    """
+    pod_obj = None
+    pvc_obj = None
+    storage_type = None
+
+    def disruptive_base(self, operation_to_disrupt, resource_to_delete):
+        """
+        Base function for PVC disruptive tests
+        """
+        DISRUPTION_OPS.set_resource(resource=resource_to_delete)
+        self.pvc_obj = helpers.create_pvc(
+            sc_name=self.sc_obj.name, wait=False
+        )
+        if operation_to_disrupt == 'create_pvc':
+            DISRUPTION_OPS.delete_resource()
+        self.pvc_obj.reload()
+        assert self.pvc_obj.ocp.wait_for_resource(condition=constants.STATUS_BOUND, resource_name=self.pvc_obj.name)
+
+        self.pod_obj = helpers.create_pod(
+            interface_type=constants.CEPHBLOCKPOOL, pvc=self.pvc_obj.name, wait=False
+        )
+        if operation_to_disrupt == 'create_pod':
+            DISRUPTION_OPS.delete_resource()
+        self.pod_obj.reload()
+
+        assert self.pod_obj.ocp.wait_for_resource(condition=constants.STATUS_RUNNING, resource_name=self.pod_obj.name)
+        self.pod_obj.run_io(storage_type=self.storage_type, size='1G')
+        if operation_to_disrupt == 'run_io':
+            DISRUPTION_OPS.delete_resource()
+
+        self.pod_obj.delete()
+        self.pvc_obj.delete()
```

Comment:
@ebenahar  should we wait for pod_obj to get deleted before deleting the pvc ?

Reply:
@nehaberry so far I haven't encountered issues with the pod staring in Terminating. I believe this should be addressed as part of https://github.com/red-hat-storage/ocs-ci/issues/372

Reply:
ok. sure.